File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/java/org/acme/spring/data/jpa/FruitResource.java
package org.acme.spring.data.jpa;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/greeting")
public class FruitResource {

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello";
    }
}

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/kotlin/src/main/kotlin/org/acme/resteasyjackson/FruitResource.kt
package org.acme.resteasyjackson

import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("/resteasy-jackson/fruits")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class FruitResource {

    private val fruits: MutableSet<Fruit> = mutableSetOf(
            Fruit("Apple", "Winter fruit"),
            Fruit("Pineapple", "Tropical fruit"),
            Fruit("Strawberry", null)
    )

    @GET
    fun list() = fruits

    @POST
    fun add(fruit: Fruit): Set<Fruit> {
        fruits.add(fruit)
        return fruits
    }

    @DELETE
    fun delete(fruit: Fruit): Set<Fruit> {
        fruits.removeIf { existingFruit: Fruit -> existingFruit.name.contentEquals(fruit.name) }
        return fruits
    }
}